llama_index/agent/openai_agent.py

```diff
@@ -236,7 +240,10 @@ async def _acall_function(self, tools: List[BaseTool], function_call: dict) -> N
                 tools, function_call, verbose=self._verbose
             )
             event.on_end(payload={EventPayload.FUNCTION_OUTPUT: str(tool_output)})
-        self.sources.append(tool_output)
+        if isinstance(tool_output, AgentChatResponse):
```

Comment:
Isn't the tool output always a `ToolOutput` object, i.e. Line 54

In what case would this not be true? 👀

Reply:
@logan-markewich Sorry for this! No idea what I was thinking, was just plain stupid. Fixed it, the correction to `BaseAgent` query methods suffices.